### PR TITLE
feat(@meso-network/meso-js): ✨ Introduce Inline integration

### PR DESCRIPTION
**Pull request by kyledetella**
This introduces a new (limited release) feature to allow partners to integrate Meso in their application inside of a specified DOM node. Instead of the Meso experience being a full-viewport modal, users can transfer from directly within the partner's application.

The previous `transfer()` integration remains in tact, but there is a new `inlineTransfer` method.

Public docs coming soon.

### CODE REVIEW

File: packages/meso-js/src/frame.ts [hunk 1]
```diff
@@ -17,20 +23,61 @@ export const setupFrame = (
...
   iframe.style.border = "none";
   iframe.style.boxSizing = "border-box";
   iframe.style.backgroundColor = "transparent";
   iframe.style.colorScheme = "auto";
+  iframe.style.width = "100%";
+  iframe.style.height = "100%";
+  iframe.style.position = "fixed";
+  iframe.style.left = "0";
+  iframe.style.top = "0";
+  iframe.style.zIndex = "9999";
```
Comment (dseeto) [inline packages/meso-js/src/frame.ts:80]: Nit: can we extract out a const for shared iframe styles?

File: packages/meso-js/src/types.ts [hunk 2]
```diff
@@ -488,8 +548,22 @@ export type Message =
...
-  | { kind: MessageKind.CLOSE }
-  | { kind: MessageKind.READY }
+  | {
+      kind: MessageKind.CLOSE | MessageKind.READY;
+    }
+  | {
+      kind: MessageKind.INITIATE_MODAL_ONBOARDING;
+      payload: {
+        /**
+         * The qualified pathname (including leading `/`) in Onboarding that the user will land on once the modal is opened.         *
```
Comment (dseeto) [inline packages/meso-js/src/types.ts:558]: Nit: trailing `*`

File: packages/meso-js/src/inlineTransfer.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      reply({
+        kind: MessageKind.RESUME_INLINE_FRAME,
+        payload: message.payload,
+      });
+    }
+  });
+
+  return {
+    destroy: () => {
+      frame.remove();
```
Comment (dseeto) [inline packages/meso-js/src/inlineTransfer.ts:105]: I think we should attempt to remove any frame in the `frameStore` as well. The partner can call `destroy` at any moment, including in the middle of onboarding.

File: packages/meso-js/src/createPostMessageBus.ts [hunk 4]
```diff
@@ -112,24 +120,65 @@ export const createPostMessageBus = (
...
     ) {
       return;
     }
 
-    handlers.get(parsedMessageResult.value.kind)?.forEach((handlerFn) => {
-      handlerFn(parsedMessageResult.value, (message: Message) => {
-        postMessageWindow.postMessage(message, event.origin);
+    const handlers = handlerStore.get(parsedMessageResult.value.kind);
+
+    if (handlers) {
```
[the remark this answers is not in the document]
Comment (kyledetella) [reply packages/meso-js/src/createPostMessageBus.ts]: Good call. Updated

File: packages/meso-js/src/types.ts (hunk 2 above)
Comment (kyledetella) [reply packages/meso-js/src/types.ts]: Nice catch

File: packages/meso-js/src/frame.ts (hunk 1 above)
Comment (kyledetella) [reply packages/meso-js/src/frame.ts]: Updated

File: packages/meso-js/src/inlineTransfer.ts (hunk 3 above)
Comment (kyledetella) [reply packages/meso-js/src/inlineTransfer.ts:105]: Good call

**Review by dseeto — APPROVED** [review]

**Pull request merged**